include/xtensor-zarr/xzarr_aws_store.hpp

```diff
@@ -96,17 +95,19 @@ namespace xt
         return bytes;
     }
 
-    void xzarr_aws_stream::operator=(const std::vector<char>& value)
+    const xzarr_aws_stream& xzarr_aws_stream::operator=(const std::vector<char>& value) const
```

Comment:
the method should not be const (since you're assigning to the object, it will change its state), and shoud return a non-const reference:

```cpp
zarr_aws_stream&  xzarr_aws_stream::operator=(const std::vector<char>& value)
{
    //...
    return *this;
}
```

Reply:
But it doesn't modify its state (i.e. none of its members are changed). Ultimately it only change an object in the cloud, but that's independent from the C++ class, isn't it?

Reply:
Indeed, but the fact that it changes an object one the cloud is an implementation detail. The "logical" state of the object is changed:

```cpp
xzarr_aws_stream stream("path1", ....);
std::string s1 = stream;
stream = "path2";
std::string s2 = stream
// => s1 != s2
```

The API should reflect logical constness, not physical constness.

Reply:
Fair enough!